test/surface_albedo.jl

```diff
@@ -0,0 +1,201 @@
+                push!(
+                    α_dir_list,
+                    ClimaAtmos.surface_albedo_direct(albedo_model)(
+                        FT(0),
+                        cos(θ_rad),
+                        FT(wind),
+                    ),
+                )
+            end
+            # plot!(
+            #     θ_list * 180 / π,
+            #     α_dir_list,
+            #     label = "n = $n, wind = $wind",
+            #     xlabel = "θ (deg)",
+            #     ylabel = "α_dir",
+            #     yaxis = :log,
+            #     yticks = ([0.001, 0.01, 0.1, 1], ["0.001", "0.01", "0.1", "1"]),
+            #     ylims = (0.001, 1),
+            #     legend = :topleft,
+            # )
+            push!(α_dir_list_all, α_dir_list)
+        end
+    end
+    # png(plot, "direct_albedo_fig2.png")
+    @test all(
+        isapprox.(
+            [α_dir_list_all[1], α_dir_list_all[2], α_dir_list_all[3]],
+            [visually_confirmed_1, visually_confirmed_2, visually_confirmed_3];
+            atol = sqrt(eps(FT)),
+        ),
+    )
+
+    # DIFFUSE
+    # visually confirmed values from Jin et al 2011
+    visually_confirmed_1 = [
+        0.04262783315710594,
+        0.038176685155394165,
+        0.03569722128776395,
+        0.033763277825400756,
+        0.03212188982700216,
+        0.03067043726101321,
+    ]
+    visually_confirmed_2 = [
+        0.06496775834930599,
+        0.06015739012695725,
+        0.057477826519383775,
+        0.055387808355172354,
+        0.053613955707494776,
+        0.052045366818292554,
+    ]
+    visually_confirmed_3 = [
+        0.08252056079082061,
+        0.0774279480438555,
+        0.07459116317553006,
+        0.07237851485996,
+        0.07050058282082186,
+        0.06883995822772186,
+    ]
+
+    n_list = [1.2, 1.34, 1.45]
+    u_list = collect(0:5:25)
+    # plot = Plots.plot()
+    α_diff_list_all = []
+    for n in n_list
+        α_diff_list = []
+        for wind in u_list
+            albedo_model = ClimaAtmos.RegressionFunctionAlbedo{FT}(n = n)
+            push!(
+                α_diff_list,
+                ClimaAtmos.surface_albedo_diffuse(albedo_model)(
+                    FT(0),
+                    cos(30 * π / 180),
+                    FT(wind),
+                ),
+            )
+        end
+        # plot!(
+        #     u_list,
+        #     α_diff_list,
+        #     label = "n = $n",
+        #     xlabel = "wind (m/s)",
+        #     ylabel = "α_diff",
+        #     yticks = (collect(0:0.02:0.1), collect(0:0.02:0.1)),
+        #     ylims = (0, 0.1),
+        # )
+        push!(α_diff_list_all, α_diff_list)
+    end
+    # png(plot, "diffuse_albedo_fig4.png")
+
+    @test all(
+        isapprox.(
+            [α_diff_list_all[1], α_diff_list_all[2], α_diff_list_all[3]],
+            [visually_confirmed_1, visually_confirmed_2, visually_confirmed_3];
+            atol = sqrt(eps(FT)),
+        ),
+    )
```

Comment:
Instead of creating lists of lists and testing afterwards, can you put the test in the for loop?

Reply:
Sure :)